parser_test.go

```diff
@@ -96,6 +96,47 @@ func TestParseStruct(t *testing.T) {
 			},
 		},
 	}
+	hiddenNestedCfg := &struct {
+		Sub struct {
+			Name string
+			Sub2 struct {
+				Name string
+			}
+		} `flag:",hidden"`
+		Sub3 struct {
+			Name string
+		}
+	}{
+		Sub: struct {
+			Name string
+			Sub2 struct {
+				Name string
+			}
+		}{
+			Name: "name_value",
+			Sub2: struct{ Name string }{
+				Name: "other_value",
+			},
+		},
+		Sub3: struct{ Name string }{
+			Name: "name_value",
+		},
+	}
+	deprecatedNestedCfg := &struct {
+		Sub struct {
+			Name string
+		} `flag:",deprecated"`
+		Sub2 struct {
+			Name string
+		}
+	}{
```

Comment:
Should this also demonstrate a nested config like `hiddenNestedCfg` above?

```go
&struct {
	Sub struct {
		Name string
		Sub2 struct {
			Name string
		}
	} `flag:",deprecated"`
	Sub3 struct {
		Name string
	}
}
```

Reply:
Good shout, I'll take a look tonight. Thanks